packages/renderer/src/lib/ContainerDetailsLogs.svelte

```diff
@@ -90,14 +93,33 @@ async function refreshTerminal() {
   // call fit addon each time we resize the window
   window.addEventListener('resize', () => {
     if (currentRouterPath === `/containers/${container.id}/logs`) {
-      fitAddon.fit();
+      termFit.fit();
     }
   });
-  fitAddon.fit();
+  termFit.fit();
+}
+
+function resizeTerminal() {
+  if (logsTerminal) {
+    termFit.fit();
+  }
 }
+
 onMount(async () => {
   refreshTerminal();
 });
+
+onMount(() => {
+  // Resize the terminal each time we change the div size
+  const resizeObserver = new ResizeObserver(entries => {
+    resizeTerminal();
+  });
+
+  resizeObserver.observe(logsXtermDiv);
+
+  // Cleanup the observer
+  return () => resizeObserver.unobserve(logsXtermDiv);
```

Comment:
You can also move the code to onDestroy() method to make it more visible (your code is still valid as onMount can return a function but it's less explicit)